### PR TITLE
Template and dependency updates

### PR DESCRIPTION
**Pull request by dontseyit**
This PR updates the microservice template and outdated dependencies.

### CODE REVIEW

**Comment by dontseyit** [comment]
Problems on my initial PR have fixed by @TheByronHimes the pydantic v2 migration as well as dependency updates should be done now.

**Review by KerstenBreuer — COMMENTED** [review]
Giving this PR to our pydantic v2 chief warrior @dontseyit for review :)

File: src/metldata/artifacts_rest/config.py [hunk 1]
```diff
@@ -30,7 +31,7 @@ class ArtifactsRestConfig(BaseSettings):
     )
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/artifacts_rest/config.py:33]:
```suggestion
```
Should come out

File: src/metldata/artifacts_rest/models.py [hunk 2]
```diff
@@ -74,7 +75,7 @@ class ArtifactInfo(BaseModel):
     )
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/artifacts_rest/models.py:77]:
```suggestion
```

File: src/metldata/builtin_transformations/aggregate/config.py [hunk 3]
```diff
@@ -37,7 +38,7 @@ class AggregationOperation(BaseModel):
     function: type[AggregationFunction]
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/builtin_transformations/aggregate/config.py:40]:
```suggestion
```

File: src/metldata/builtin_transformations/merge_slots/config.py [hunk 4]
```diff
@@ -44,7 +47,7 @@ class SlotMergingConfig(BaseSettings):
     )
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/builtin_transformations/merge_slots/config.py:49]:
```suggestion
```

File: src/metldata/event_handling/artifact_events.py [hunk 5]
```diff
@@ -32,7 +32,7 @@ class ArtifactEventConfig(BaseModel):
     )
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/event_handling/artifact_events.py:34]:
```suggestion
```

File: src/metldata/load/collect.py [hunk 6]
```diff
@@ -41,7 +41,7 @@ class ArtifactCollectorConfig(ArtifactEventConfig):
     )
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/load/collect.py:43]:
```suggestion
```

File: src/metldata/submission_registry/models.py [hunk 7]
```diff
@@ -106,7 +106,7 @@ def current_status(self) -> SubmissionStatus:
         return sorted_history[-1].new_status
 
     # pylint: disable=no-self-argument
```
Comment (TheByronHimes) [inline src/metldata/submission_registry/models.py:108]:
```suggestion
```

**Review by TheByronHimes — APPROVED** [review]
There are some pylint ignores that should be taken out, but it's of course not a pressing issue.

**Pull request merged**